test/dnf/test_product_id.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+import unittest
+import os
+import tempfile
+from typing import Union
+
+from plugins.dnf.product_id import DnfProductManager
+
+
+class TestDnfPluginProductId(unittest.TestCase):
+
+    CACHE_FILE = "productid_repo_mapping.json"
+
+    def setUp(self) -> None:
+        self.temp_directory: Union[None, tempfile.TemporaryDirectory] = None
+
+    def tearDown(self) -> None:
+        if self.temp_directory is not None:
+            self.temp_directory.cleanup()
+
+    def _create_cache_file(self) -> str:
+        """
+        Create temporary directory and return path of cache file
+        """
+        self.temp_directory = tempfile.TemporaryDirectory()
+        return os.path.join(self.temp_directory.name, self.CACHE_FILE)
+
+    def _create_cache_file_read_only(self) -> str:
+        """
+        Create read-only temporary directory and return path of cache file
+        """
+        self.temp_directory = tempfile.TemporaryDirectory()
+        # Make directory read only
+        os.chmod(self.temp_directory.name, 0x400)
+        return os.path.join(self.temp_directory.name, self.CACHE_FILE)
+
+    def test_create_cache_file(self):
+        """
+        Test that it is possible to create cache file
+        """
+        cache_file = self._create_cache_file()
+        DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        data = DnfProductManager._read_cache_file(cache_file)
+        self.assertEqual(data, {"foo": "bar"})
+
+    def test_create_cache_file_in_read_only_directory(self):
```

Comment:
this test will likely fail as root, as root can write everything, see #3104; please add a skip for that

Reply:
Good point.